### PR TITLE
Box Rotation

### PR DESCRIPTION
**Pull request by AlexandruPopovici**
## Description & motivation
Objects of speckle type `Box` now get their base plane basis applied as rotation

## Checklist:

- [ ] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [ ] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [ ] My commits are related to the pull request and do not amend unrelated code or documentation.
- [ ] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

File: packages/viewer/src/modules/loaders/Speckle/SpeckleGeometryConverter.ts [hunk 1]
```diff
@@ -397,7 +397,19 @@ export class SpeckleGeometryConverter extends GeometryConverter {
...
     const conversionFactor = getConversionFactor(node.raw.units)
 
-    const move = this.PointToVector3(node.raw.basePlane.origin)
+    const T = new Matrix4()
+    const R = new Matrix4()
+
+    if (node.raw.basePlane) {
+      T.setPosition(this.PointToVector3(node.raw.basePlane.origin))
+
+      R.makeBasis(
```
[the remark this answers is not in the document]
Comment (AlexandruPopovici) [reply packages/viewer/src/modules/loaders/Speckle/SpeckleGeometryConverter.ts:406]: It will not throw.

**Review by clairekuang — APPROVED** [review]
approved, with an extra comment

File: packages/viewer/src/modules/loaders/Speckle/SpeckleGeometryConverter.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (clairekuang) [reply packages/viewer/src/modules/loaders/Speckle/SpeckleGeometryConverter.ts:406]: yeah, i just mean we don't have any validation in place to guarantee those vectors will be perp

[the remark this answers is not in the document]
Comment (AlexandruPopovici) [reply packages/viewer/src/modules/loaders/Speckle/SpeckleGeometryConverter.ts]: @clairekuang I've added the orthogonal check

**Review by clairekuang — APPROVED** [review]
looks good!

**Pull request merged**